### ISSUE TITLE
Snap: Set outbound Proxy

### ISSUE THREAD

**Issue by irvingwa** [RocketChat/Rocket.Chat#4780]
Hey all running the latest rocket.chat with a snap install. What is the best way to set both proxy and no proxy lists with the snap install?

**Comment by geekgonecrazy** [comment]
@irvingwa just so I understand. You want the ability to set an outbound proxy for Rocket.Chat?

**Comment by Velociround** [comment]
I don't know what he wants, but my Rocket Chat server uses the internet via a corporate proxy. I need RocketChat to use the same proxy as the system, or to setup proxy inside RocketChat so that it can connect to the internet. My server does wget and yum install fine through proxy but RocketChat gets

`[34mI20170112-15:56:56.633(-2) app.js:471 [31mSystem ➔ error Error sending push to gateway (6 try) -> { [Error: connect ECONNREFUSED 45.55.199.111:443]   code: 'ECONNREFUSED',   errno: 'ECONNREFUSED',   syscall: 'connect',   address: '45.55.199.111',   port: 443 }`

**Comment by zbyte** [comment]
this feature is very important for me

**Comment by holgerdell** [comment]
How do I make a rocket.chat snap instance aware of the HTTP_PROXY and HTTPS_PROXY environment variables?

Using `systemctl edit snap.rocketchat-server.rocketchat-server.service`, I can edit the environment variables to be used inside the snap container as follows:

```
[Service]
Environment=HTTP_PROXY=http://my.proxy-server.example.com:3128
Environment=HTTPS_PROXY=https://my.proxy-server.example.com:3128
```

However, rocket.chat appears to ignore this setting. Any ideas what I can do to debug this?

Possible duplicates: #5090, #5896

**Comment by geekgonecrazy** [comment]
if you edit it will be replaced any time the snap is updated.  the systemd unit file is created by snapd automatically

**Comment by holgerdell** [comment]
@geekgonecrazy 

1. I just checked, and the edit did not get replaced in the last automatic updates.
2. More importantly, the setting appears to have no effect. Can you help me solve/analyze the problem? I have no experience with snaps.

**Comment by geekgonecrazy** [comment]
I'm not sure if it honors the proxy even when not using snap.. I've seen very mixed reports

**Comment by holgerdell** [comment]
@geekgonecrazy
According to [this](https://github.com/meteor/meteor/wiki/Using-Meteor-behind-a-proxy), meteor is supposed to honor the HTTP_PROXY and HTTPS_PROXY environment variables. Is there a way I can issue the meteor update command inside a snap?

**Comment by geekgonecrazy** [comment]
Meteor actually does not run inside the snap.

Meteor is a node.js framework.  When we cut a release we do a meteor build which produces a standalone meteor application.

We will be adding a way to easily set these environment variables in the near future.

**Comment by holgerdell** [comment]
Sounds good, thanks!

If possible, the snap should ideally just honor the host system's proxy variables in /etc/environment

**Comment by geekgonecrazy** [comment]
I'm not sure if that is possible just by the very nature of snaps.  Isolation / privacy / security are all kind of the name of the game for snaps.  One of the reasons we chose to use it.

**Comment by reetp** [comment]
Not sure if this gets closed by:
https://github.com/RocketChat/Rocket.Chat/pull/12351

**Comment by holgerdell** [comment]
@reetp How can I test this? Can you tell me an operation that causes the rocket.chat server to request a file from the web?

**Comment by reetp** [comment]
Paste a link to a file online or a web page is probably easiest

**Comment by holgerdell** [comment]
Nothing happens when I do that.

**Comment by geekgonecrazy** [comment]
This should technically get closed by #12351 it will allow setting environment variables

**Comment by geekgonecrazy** [comment]
#12351 will allow setting of environment variables so any that will be possible with non-snap will now also be possible to set there.  Will be released with 0.73.0

**Comment by Antairez** [comment]
@geekgonecrazy Thanks for the info, are you saying that setting proxy the way just like @holgerdell did (by running `sudo systemctl edit snap.rocketchat-server.rocketchat-server.service`)will be possible after the release of 0.73.0 for snap users?



```
[Service]
Environment=HTTP_PROXY=http://proxy.example.com:8080/
Environment=HTTPS_PROXY=http://proxy.example.com:8080/
Environment=NO_PROXY='something.example.com,*.intranet.example.com'
```


Or is there a recommended way to set it properly and make sure it persists across snap refreshes?

**Comment by geekgonecrazy** [comment]
After the snap is released for 0.73.0 you will be able to set any environment variable for the snap by dropping a .env file in the common folder.  I believe the docs will be updated soon.  It will be a few days as usual before the snap release is available.

**Comment by Antairez** [comment]
> After the snap is released for 0.73.0 you will be able to set any environment variable for the snap by dropping a .env file in the common folder. I believe the docs will be updated soon. It will be a few days as usual before the snap release is available.

@geekgonecrazy Now that 0.73 is out, would you mind pointing me to some instructions on how to set-up proxy using the new way? Thank you.

**Comment by reetp** [comment]
It is in the documentation
https://rocket.chat/docs/installation/manual-installation/ubuntu/snaps/

**Comment by geekgonecrazy** [comment]
To clarify it was merged today as well as 0.73 released to stable channel today :) . But yes can now set environment variables as described in that document

**Comment by Antairez** [comment]
@reetp Thanks my friend, how do I find a list of environment variables for changing proxy servers?
